core/src/consensus.rs

```diff
@@ -978,12 +981,15 @@ impl Tower {
         self.last_switch_threshold_check.is_none()
     }
 
+    /// Performs threshold check for `slot`
+    ///
+    /// If it passes the check returns None, otherwise returns Some(fork_stake)
     pub fn check_vote_stake_threshold(
         &self,
         slot: Slot,
         voted_stakes: &VotedStakes,
         total_stake: Stake,
-    ) -> bool {
```

Comment:
I think a typed `Enum` here  saying `Failed/PassedThreshold` would be better than an `Option`. The option being Some(stake) for the failure case seems a bit nonintuitive to me